node/src/effect.rs

```diff
@@ -1749,7 +1735,11 @@ impl<REv> EffectBuilder<REv> {
             // we don't support getting the validators from before the last emergency restart
             return None;
         }
-        self.get_key_block_header_for_era_id_from_storage(era_id)
+        // Era 0 contains no blocks other than the genesis immediate switch block which can be used
+        // to get the validators for era 0.  For any other era `n`, we need the switch block from
+        // era `n-1` to get the validators for `n`.
+        let era_before = era_id.saturating_sub(1);
```

Comment:
Not required, but consider adding `EraId::predecessor()` similarly to the already existing `EraId::successor()`.

Reply:
Hmm - I think I'll leave as is thanks, since we'd be replacing a one-line call here with a call which is a little more opaque IMO.  There might be cases where we'd want `checked_sub` so we get `None` in the case of era 0.